### PR TITLE
[UPDATE] pin sonar action to commit

### PR DESCRIPTION
**Pull request by sethkfman**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

This PR pins the sonar action to a specific commit per our organizations requirements. This action has no impact on production source code and is only used for static analysis. 

**Screenshots/Recordings**

NA

**Issue**

NA

**Checklist**

* [NA] There is a related GitHub issue
* [NA] Tests are included if applicable
* [NA] Any added code is fully documented

### CODE REVIEW

File: .github/workflows/sonar.yml [hunk 1]
```diff
@@ -11,7 +11,8 @@ jobs:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 0
-      - uses: sonarsource/sonarqube-scan-action@master
+      - uses: sonarsource/sonarqube-scan-action@427bad70165a74f9d7133d48ea31b482a53ee9fa
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
           SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
+
```
Comment (Gudahtt) [inline .github/workflows/sonar.yml:18]: Nit: I think this is saying the last line has whitespace in it? Not sure

**Review by gantunesr — APPROVED** [review]
LGTM

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: .github/workflows/sonar.yml (hunk 1 above)
Comment (Gudahtt) [reply .github/workflows/sonar.yml]: In the original commit there was extra indentation added on the last line, but in this latest commit it looks like the trailing newline was removed *and* additional whitespace was added on this line :sweat_smile:

**Review by Gudahtt — APPROVED** [review]

File: .github/workflows/sonar.yml (hunk 1 above)
Comment (Gudahtt) [reply .github/workflows/sonar.yml]: We can follow up later to ensure these are covered by Prettier

**Review by leotm — APPROVED** [review]

File: .github/workflows/sonar.yml (hunk 1 above)
Comment (leotm) [reply .github/workflows/sonar.yml]: [eslint-plugin-yml](https://github.com/ota-meshi/eslint-plugin-yml) could be a good shout ^ with few fixable [rules](https://ota-meshi.github.io/eslint-plugin-yml/rules/)

and [config](https://github.com/ota-meshi/eslint-plugin-yml#book-usage) that plays nice w prettier

**Pull request merged**